packages/mobile/src/components/AccountNumber.tsx

```diff
@@ -2,31 +2,58 @@ import colors from '@celo/react-components/styles/colors.v2'
 import fontStyles from '@celo/react-components/styles/fonts.v2'
 import { getAddressChunks } from '@celo/utils/src/address'
 import React from 'react'
-import { StyleSheet, Text, View } from 'react-native'
+import { useTranslation } from 'react-i18next'
+import { Clipboard, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
+import { Namespaces } from 'src/i18n'
+import Logger from 'src/utils/Logger'
 
 interface Props {
   address: string
+  touchDisabled?: boolean
 }
 
-export default function AccountNumber({ address }: Props) {
+export default function AccountNumber({ address, touchDisabled }: Props) {
+  const { t } = useTranslation(Namespaces.accountScreen10)
+  const onPressAddress = () => {
+    if (!address.length) {
+      return
+    }
+    Clipboard.setString(address)
+    Logger.showMessage(t('addressCopied'))
+  }
   // Turns '0xce10ce10ce10ce10ce10ce10ce10ce10ce10ce10'
   // into 'ce10 ce10 ce10 ce10 ce10 ce10 ce10 ce10 ce10 ce10'
-  const formattedAddress = getAddressChunks(address).join(' ')
+  const addressChunks = ['0x', ...getAddressChunks(address)]
+
+  if (touchDisabled) {
+    return (
+      <View style={styles.container}>
+        <Text style={[styles.text, styles.topText]}>{addressChunks.slice(0, 6).join(' ')}</Text>
+        <Text style={[styles.text, styles.bottomText]}>{addressChunks.slice(6).join(' ')}</Text>
+      </View>
+    )
+  }
+
   return (
-    <View style={styles.container}>
-      <Text style={styles.text}>0x </Text>
-      <Text style={styles.text}>{formattedAddress}</Text>
-    </View>
+    <TouchableOpacity style={styles.container} onPress={onPressAddress}>
```

Comment:
not a huge deal, but in the interest of DRY you could do a ternary instead of the if statement:
`touchDisabled ? <View... : <Touchable...`

Reply:
Yep good call